packages/user/Invite/src/Invite.cpp

```diff
@@ -1,40 +1,81 @@
 #include <services/system/Accounts.hpp>
 #include <services/system/AuthAny.hpp>
 #include <services/system/AuthSig.hpp>
+#include <services/system/Credentials.hpp>
 #include <services/system/HttpServer.hpp>
 #include <services/system/Transact.hpp>
 #include <services/system/commonErrors.hpp>
-#include <services/user/AuthInvite.hpp>
 #include <services/user/Events.hpp>
 #include <services/user/Invite.hpp>
 #include <services/user/Nft.hpp>
 #include <services/user/REvents.hpp>
 #include <services/user/Tokens.hpp>
 
+#include <psibase/Actor.hpp>
 #include <psibase/Bitset.hpp>
-#include <vector>
+#include "psibase/db.hpp"
+#include "psibase/nativeTables.hpp"
+#include "psibase/serviceState.hpp"
+#include "services/user/InviteErrors.hpp"
 
 using namespace psibase;
 using namespace UserService;
 using namespace UserService::InviteNs;
 using namespace UserService::Errors;
 using namespace SystemService;
-using psio::view;
-using std::begin;
-using std::end;
-using std::find;
 using std::optional;
 using std::string;
-using std::vector;
+using std::chrono::duration_cast;
+using std::chrono::seconds;
+using std::chrono::weeks;
+
+namespace
+{
+   constexpr seconds ONE_WEEK = duration_cast<Seconds>(weeks(1));
+
+   void hookOnInvAccept(const InviteRecord& invite, AccountNumber accepter)
+   {
+      if (invite.useHooks)
+      {
+         Actor<InviteHooks>{Invite::service, invite.inviter}.onInvAccept(invite.id, accepter);
+      }
+   }
+
+   void hookOnInvDelete(const InviteRecord& invite)
+   {
+      if (invite.useHooks)
+      {
+         Actor<InviteHooks>{Invite::service, invite.inviter}.onInvDelete(invite.id);
+      }
+   }
+
+}  // namespace
 
 Invite::Invite(psio::shared_view_ptr<Action> action)
 {
    MethodNumber m{action->method()};
    if (m != MethodNumber{"init"})
    {
-      auto initRecord = Tables().open<InitTable>().get({});
+      auto initRecord = Tables(getReceiver(), KvMode::read).open<InitTable>().get({});
       check(initRecord.has_value(), UserService::Errors::uninitialized);
    }
+
+   if (getSender() == Credentials::CREDENTIAL_SENDER)
+   {
+      if (m != "createAccount"_m)
+      {
+         string error = Credentials::CREDENTIAL_SENDER.str() + canOnlyCallCreateAccount.data();
+         abortMessage(error);
+      }
```

Comment:
This looks error prone. I think it would be better for Credentials::create to take a whitelist of actions, so it's impossible to accidentally forget.